lib/elixometer.ex

```diff
@@ -122,11 +122,12 @@ defmodule Elixometer do
     end
   end
 
-  defp build_timer_body(%Timer{} = timer_data) do
-    quote do
-      timed(unquote(timer_data.key), unquote(timer_data.units)) do
-        unquote(timer_data.body)
-      end
+  defp build_timer_body(%Timer{body: [do: block]} = timer) do
+    build_timer_body(%{timer | body: block})
```

Comment:
This will get the wrong body in an extreme edge case when if body of the function is of the form `[do: ...]` because keywords are keywords in the AST. For example:
```elixir
def keyword_body(), do: [do: :value]
```
This is very unfortunate behaviour on Elixir's part :( and I think forces a version check (unsure how to do feature check for this), with something like:
```elixir
defp get_body(%Timer{body: body}) do
  case Version.compare(System.version(), "1.5.0-rc") do
    :lt -> [do: body]
    _ when is_nil(body) -> raise "can not time bodyless function"
    _ -> body
  end
end
```